heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/service/OAuthInterceptorService.java

```diff
@@ -146,27 +160,17 @@ public void executeInterceptor(TypeOAuth typeOAuth, String privateKey, int timeA
             .checkThrow(Objects.isNull(oAuthRequest.getGrantType()) || oAuthRequest.getGrantType().isEmpty(), ExceptionMessage.GRANT_TYPE_NOT_FOUND);
 
         HttpServletRequest request = context.getRequest();
-        String body = "";
-
-        try {
-
-            body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
-        } catch (IOException e) {
-            log.error(e.getMessage(), e);
-        }
+        String body = getBodyRequest(request);
 
         timeAccessToken = (timeAccessToken <= 0) ? TIME_ACCESS_TOKEN : timeAccessToken;
         timeRefreshToken = (timeRefreshToken <= 0) ? TIME_REFRESH_TOKEN : timeRefreshToken;
 
-        String clientId = request.getHeader("client_id");
-        String accessToken = request.getHeader("access_token");
-
         switch (typeOAuth) {
             case GENERATE:
                 runGenerate(oAuthRequest, privateKey, timeAccessToken, timeRefreshToken, body);
```

Comment:
If `runAuthorize` was refactored to receive `request`, I think `runGenerate` should have too. This way we avoid `getBodyRequest` be request twice everytime that `runAuthoriza` is processed (1 one time in `executeInterceptor` and 1 one time in `passwordFlow`, `implicitFlow` or `refreshFlow`, depending of the context